aspnetcore/fundamentals/websockets.md

```diff
@@ -59,43 +59,76 @@ If you don't want to wait for SignalR Core, you can use WebSockets directly now.
+The following example is from later in the `Configure` method:
 
 [!code-csharp[](websockets/sample/Startup.cs?name=AcceptWebSocket&highlight=7)]
 
 A WebSocket request could come in on any URL, but this sample code only accepts requests for `/ws`.
 
 ### Send and receive messages
 
-The `AcceptWebSocketAsync` method upgrades the TCP connection to a WebSocket connection and gives you a [WebSocket](https://docs.microsoft.com/dotnet/core/api/system.net.websockets.websocket) object. Use the WebSocket object to send and receive messages.
+The `AcceptWebSocketAsync` method upgrades the TCP connection to a WebSocket connection and provides a [WebSocket](/dotnet/core/api/system.net.websockets.websocket) object. Use the WebSocket object to send and receive messages.
 
-The code shown earlier that accepts the WebSocket request passes the `WebSocket` object to an `Echo` method; here's the `Echo` method. The code receives a message and immediately sends back the same message. It stays in a loop doing that until the client closes the connection. 
+The code shown earlier that accepts the WebSocket request passes the `WebSocket` object to an `Echo` method. The code receives a message and immediately sends back the same message. Messages are sent and received in a loop until the client closes the connection:
 
 [!code-csharp[](websockets/sample/Startup.cs?name=Echo)]
 
-When you accept the WebSocket before beginning this loop, the middleware pipeline ends.  Upon closing the socket, the pipeline unwinds. That is, the request stops moving forward in the pipeline when you accept a WebSocket, just as it would when you hit an MVC action, for example.  But when you finish this loop and close the socket, the request proceeds back up the pipeline.
+When accepting the WebSocket connection before beginning the loop, the middleware pipeline ends. Upon closing the socket, the pipeline unwinds. That is, the request stops moving forward in the pipeline when the WebSocket is accepted, just as it would when an MVC action it hit. When the loop is finished and the socket is closed, the request proceeds back up the pipeline.
+
+## IIS/IIS Express support
+
+Windows Server 2012 or later and Windows 8 or later with IIS/IIS Express 8 or later has support for the WebSocket Protocol.
+
+To enable support for the WebSocket Protocol on Windows Server 2012 or later:
+
+1. Use the **Add Roles and Features** wizard from the **Manage** menu or the link in **Server Manager**.
+1. Select **Role-based or Feature-based Installation**. Select **Next**.
+1. Select the appropriate server (the local server is selected by default). Select **Next**.
+1. Expand **Web Server (IIS)** in the **Roles** tree, expand **Web Server**, and then expand **Application Development**.
+1. Select **WebSocket Protocol**. Select **Next**.
+1. If additional features aren't needed, select **Next**.
+1. Select **Install**.
+1. When the installation completes, select **Close** to exit the wizard.
```

Comment:
Screenshots might be nice, those dialogs can be confusing

Reply:
We're avoiding screenshots except in tutorials. I think it's due to "documentation debt" lol ... the corollary to "technical debt." In a future update to a reference doc, we must not only update the text but re-shoot the screenshots. It's rather time-consuming. Fortunately for these steps, it's mostly just opening nodes until the feature checkbox appears.

Reply:
Ok, I buy that. Thanks for clarification.